server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -876,11 +877,23 @@ public Profilers getProfilers() {
     }
 
     /**
-     * Returns concurrent segment search status for the search context
+     * Returns concurrent segment search status for the search context. This should only be used after request parsing, during which requestShouldUseConcurrentSearch will be set.
      */
     @Override
-    public boolean isConcurrentSegmentSearchEnabled() {
-        return useConcurrentSearch;
+    public boolean shouldUseConcurrentSearch() {
+        assert requestShouldUseConcurrentSearch != null : "requestShouldUseConcurrentSearch must be set";
+        return concurrentSearchSettingsEnabled && requestShouldUseConcurrentSearch;
+    }
+
+    /**
+     * Evaluate if parsed request supports concurrent segment search
+     */
+    public void evaluateRequestShouldUseConcurrentSearch() {
+        if (aggregations() != null && aggregations().factories() != null) {
+            requestShouldUseConcurrentSearch = aggregations().factories().allFactoriesSupportConcurrentSearch();
+        } else {
+            requestShouldUseConcurrentSearch = concurrentSearchSettingsEnabled;
```

Comment:
we can set ` requestShouldUseConcurrentSearch` to `true` by default instead of `concurrentSearchSettingsEnabled` as `shouldUseConcurrentSearch` is checking for both.